torch/csrc/api/include/torch/nn/options/linear.h

```diff
@@ -42,6 +43,33 @@ struct TORCH_API FlattenOptions {
 
 // ============================================================================
 
+/// Options for the `Unflatten` module.
+///
+/// Note: If input tensor is named, use dimname and namedshape arguments.
+///
+/// Example:
+/// ```
+/// Unflatten unnamed_model(UnflattenOptions(0, {2, 2}));
+/// Unflatten named_model(UnflattenOptions("B", {{"B1", 2}, {"B2", 2}}));
+/// ```
+struct TORCH_API UnflattenOptions {
+  typedef std::vector<std::pair<std::string, int64_t>> namedshape_t;
+
+  UnflattenOptions(int64_t dim, std::vector<int64_t> sizes);
+  UnflattenOptions(std::string dimname, namedshape_t namedshape);
```

Comment:
Need a constructor of UnflattenOptinoos(const char*, namedshape_t) if you want to do something like UnflattenOptions("A", ......